### PR TITLE
Map preview thumbnail fix

### PR DESCRIPTION
**Pull request by mbektash**

### CODE REVIEW

**Review by vishalya — COMMENTED** [review]
Does get_auth_token generate it? Also how will work for the existing users?

**Comment by tylerparsons** [comment]
Have you tested a user auto creation with this change? I thought `app/models/carto/user_creation.rb` was the code that actually handles user auto creations - you might need to touch that as well.

File: lib/tasks/setup.rake [hunk 1]
```diff
@@ -82,6 +84,8 @@ DESC
       user.save
       puts "User #{user.username} created successfully"
 
+      user.get_auth_token
```
Comment (vishalya) [inline lib/tasks/setup.rake]: Should we have it before save, that way it can be stored, but it depends on what get_auth_token does?

**Review by vishalya — CHANGES_REQUESTED** [review]

File: lib/tasks/setup.rake (hunk 1 above)
Comment (mbektash) [reply lib/tasks/setup.rake]: get_auth_token calls save, so it can be after user.save. no issues there.

**Comment by mbektash** [comment]
@tylerparsons build function in user_account_creator.rb should be covering that case. havent tested auto creation yet.

File: app/controllers/carto/api/visualizations_controller.rb [hunk 2]
```diff
@@ -281,13 +281,18 @@ def list
         tags = params.fetch(:tags, '').split(',')
         tags = nil if tags.empty?
         is_common_data_user = common_data_user && user_id == common_data_user.id
+        is_maps = types.count == 1 && types[0] == "'derived'"
```
Comment (gfiorav) [inline app/controllers/carto/api/visualizations_controller.rb]: Minor:

Maybe you could just check the whole thing:
```ruby
is_maps = types == ["'derived'"]
```

If you do that then you can remove the extra variable and just do this below:

```ruby
if types == ["'derived'"]
  user_join = 'LEFT JOIN users AS u ON u.id=v.user_id'
  auth_tokens_column = ', ARRAY[u.auth_token] AS auth_tokens'
end
```

**Review by gfiorav — CHANGES_REQUESTED** [review]
I left a minor comment, and another one about using a `before_create :get_auth_tokens` in the `user` models.

I also want to add a little request: Could you put a comment on top of `list` the method with the limitations? Something like:

```ruby
# NOTE: Only user auth_tokens are included due to speed limitations. May not work with shared maps (missing organisation token).
```

That will make housekeeping a lot easier in the future :)

File: lib/tasks/setup.rake (hunk 1 above)
Comment (gfiorav) [reply lib/tasks/setup.rake]: Maybe we can add it as a `before_create` filter in `app/models/user.rb` and `app/models/carto/user.rb`

```ruby
before_create get_auth_tokens
```

That should do and should work for all cases. What do you think?

File: app/models/user.rb [hunk 3]
```diff
@@ -217,6 +217,7 @@ def before_create
     super
     self.database_host ||= ::Rails::Sequel.configuration.environment_for(Rails.env)['host']
     self.api_key ||= self.class.make_token
+    populate_auth_token
```
Comment (gfiorav) [inline app/models/user.rb]: Does this method exist in this class?

**Review by gfiorav — APPROVED** [review]
Added a small comment but otherwise :+1:

**Pull request merged**